util/xmlparser.py

```diff
@@ -49,21 +49,22 @@ def int_of_optional_subtag(self, tag, name, namespaces=None):
         return int(v)
 
     def process_all(self, xml, xpath, namespaces=None, handler=None, parser=None):
-        # XMLParser can handle most characters and entities that are
-        # invalid in XML but it will stop processing a document if it
-        # encounters the null character. Remove that character
-        # immediately and XMLParser will handle the rest.
-        xml = xml.replace(b"\x00", "")
         if not parser:
             parser = etree.XMLParser(recover=True)
         if not handler:
             handler = self.process_one
         if isinstance(xml, basestring):
             root = None
-            exception = None
```

Comment:
This variable should have been removed by my previous branch along with that big chunk of code.